enterprise/server/remote_execution/dirtools/dirtools.go

```diff
@@ -307,6 +339,20 @@ func UploadTree(ctx context.Context, env environment.Env, dirHelper *DirHelper,
 		return uploadableFile.FileNode(), nil
 	}
 
+	uploadSymlinkFn := func(parentDir string, target string, info os.FileInfo) (*repb.SymlinkNode, error) {
+		uploadableSymlink, err := newSymlinksToUpload(instanceName, digestFunction, parentDir, info, target)
+		if err != nil {
+			return nil, err
+		}
+		filesToUpload = append(filesToUpload, uploadableSymlink)
```

Comment:
The current approach is to treat each symlink like a normal file by appending it to `filesToUpload`

In `func uploadFiles()` where we consume this `filesToUpload` slice, we check whether the file entry is a directory or not `uploadableFile.dir == nil` before adding it to our `interfaces.FileCache` where the file would be hardlinked into the cache dir.

Because a symlink is just a file with a different set of permission and content being the Target it should be compatible with FileCache's hardlink approach.